### ISSUE TITLE
Git storage does not support file rename

### ISSUE THREAD

**Issue by La0** [Requarks/wiki#1358]
**Describe the bug**
When renaming a file in a remote git repository (used as main storage source), the old file remains available in the wiki frontend, and the new file is not available.

**To Reproduce**
1. `git mv a.md b.md`
2. `git commit -a -m 'move a to b'`
3. `git push`
4. wait for wiki to refresh content, nothing happens (look at logs :smile: )

**Expected behavior**
`/a` should not be available, whereas `/b` should be available.

**Host Info (please complete the following information):**
 - OS: Docker (on Debian stable)
 - Wiki.js version: 2.0.12
 - Database engine: postgres 12.1

**Additional context**
I got that warning in the docker logs, the last line looks weird, it's like if the wiki worker tried to process the filename `/wiki/data/repo/a.md => b.md` instead of `/wiki/data/repo/b.md`

```
2020-01-06T21:49:11.876Z [MASTER] info: (STORAGE/GIT) Performing pull rebase from origin on branch master...
2020-01-06T21:49:13.430Z [MASTER] info: (STORAGE/GIT) Performing push to origin on branch master...
2020-01-06T21:49:14.825Z [MASTER] warn: (STORAGE/GIT) Failed to process a.md => b.md
2020-01-06T21:49:14.825Z [MASTER] warn: ENOENT: no such file or directory, open '/wiki/data/repo/a.md => b.md'
```

**Comment by akaRem** [comment]
https://docs.requarks.io/guide/pages#single-character-pages

All single character paths are reserved to access various parts of the Wiki:

**Comment by bmann** [comment]
@akaRem pretty sure that's just example file names.

I think I'm seeing the same thing.

I created a new "2002" folder, and moved all files that were 2002-* into it, and am getting:

```
2020-05-20T08:14:46.882083+00:00 app[web.1]: 2020-05-20T08:14:46.882Z [MASTER] [33mwarn[39m: (STORAGE/GIT) Failed to process archive/{ => 2002}/2002-12-23-boxes-and-arrows-beauty-is-only-screen-deep-sarah-horton.md
2020-05-20T08:14:46.882216+00:00 app[web.1]: 2020-05-20T08:14:46.882Z [MASTER] [33mwarn[39m: ENOENT: no such file or directory, open '/wiki/data/repo/archive/{ => 2002}/2002-12-23-boxes-and-arrows-beauty-is-only-screen-deep-sarah-horton.md'
```

There is no "2002" folder in the WikiJS front end, and the files are still accessible at their original paths.

Which cache can I purge so this is re-read and displays correctly? What is the correct way to move / rename files via git so it is reflected in WikiJS?

BTW: the reason I'm making this move is because I have too many files in one folder, so the front end browse navigation really slows down.

**Comment by spyesx** [comment]
I've beed working on reorganising our wiki especially from a local clone of the repository. Then I pushed all changes and I can see the same issue here.

- renamed files are not available
- moved files are not available at their new path but still at their old path
- removed items are still available

I've cleared all caches possible from the UI. `/data/repo` is perfectly sync as I can see the same structure. (That's reassuring 😄 )  "Rerender All Pages" and "Rebuild Page Tree" from Utilities don't change anything.

I thought it could come from inner links so we removed them all to switch to a "Browse" navigation only. No change.

Is there a way to rebuild the entire wiki tree from the repo?

I'd be happy to provide more informations to help if you have specific questions.
